File: frontend/src/api/siem.ts
import apiClient from "./client";

export interface SiemConnector {
  id: string;
  name: string;
  siem_type: string;
  base_url: string;
  webhook_secret: string;
  is_active: boolean;
  last_event_at: string | null;
  created_at: string;
}

export interface SiemConnectorCreate {
  name: string;
  siem_type: string;
  base_url: string;
  config?: Record<string, unknown>;
  webhook_secret?: string;
}

export interface SiemAlert {
  id: string;
  connector_id: string;
  source_rule_id: string | null;
  severity: string;
  title: string;
  description: string | null;
  affected_host: string | null;
  source_ip: string | null;
  normalized_at: string;
  playbook_triggered: boolean;
}

export const siemApi = {
  listConnectors: () =>
    apiClient.get<SiemConnector[]>("/siem").then((r) => r.data),

  createConnector: (data: SiemConnectorCreate) =>
    apiClient.post<SiemConnector>("/siem", data).then((r) => r.data),

  getConnector: (id: string) =>
    apiClient.get<SiemConnector>(`/siem/${id}`).then((r) => r.data),

  updateConnector: (id: string, data: SiemConnectorCreate) =>
    apiClient.patch<SiemConnector>(`/siem/${id}`, data).then((r) => r.data),

  deleteConnector: (id: string) =>
    apiClient.delete(`/siem/${id}`),

  listAlerts: (params?: { severity?: string; triggered_only?: boolean; limit?: number }) =>
    apiClient.get<SiemAlert[]>("/siem/alerts/list", { params }).then((r) => r.data),

  ingest: (connectorId: string, payload: Record<string, unknown>) =>
    apiClient.post<{ alert_id: string; severity: string; title: string }>(
      `/siem/${connectorId}/ingest`,
      payload
    ).then((r) => r.data),
};

File: backend/app/models/siem.py
"""ORM models para F37 — Integrações SIEM."""
from __future__ import annotations

import secrets
from datetime import datetime
from uuid import UUID, uuid4

from sqlalchemy import Boolean, ForeignKey, String, Text, TIMESTAMP
from sqlalchemy.dialects.postgresql import JSONB, UUID as PG_UUID
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.sql import func

from app.database import Base


class SiemConnector(Base):
    __tablename__ = "siem_connectors"

    id: Mapped[UUID] = mapped_column(PG_UUID(as_uuid=True), primary_key=True, default=uuid4)
    tenant_id: Mapped[UUID] = mapped_column(PG_UUID(as_uuid=True), ForeignKey("tenants.id", ondelete="CASCADE"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    siem_type: Mapped[str] = mapped_column(String(30), nullable=False)   # wazuh|splunk|sentinel|log360|qradar
    base_url: Mapped[str] = mapped_column(Text, nullable=False)
    config_encrypted: Mapped[str | None] = mapped_column(Text, nullable=True)
    webhook_secret: Mapped[str] = mapped_column(String(64), nullable=False, default=lambda: secrets.token_hex(32))
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    last_event_at: Mapped[datetime | None] = mapped_column(TIMESTAMP(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(TIMESTAMP(timezone=True), server_default=func.now(), nullable=False)


class SiemAlert(Base):
    __tablename__ = "siem_alerts"

    id: Mapped[UUID] = mapped_column(PG_UUID(as_uuid=True), primary_key=True, default=uuid4)
    tenant_id: Mapped[UUID] = mapped_column(PG_UUID(as_uuid=True), ForeignKey("tenants.id", ondelete="CASCADE"), nullable=False, index=True)
    connector_id: Mapped[UUID] = mapped_column(PG_UUID(as_uuid=True), ForeignKey("siem_connectors.id", ondelete="CASCADE"), nullable=False)
    source_rule_id: Mapped[str | None] = mapped_column(Text, nullable=True)
    severity: Mapped[str] = mapped_column(String(20), nullable=False, default="medium")
    title: Mapped[str] = mapped_column(Text, nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    affected_host: Mapped[str | None] = mapped_column(Text, nullable=True)
    source_ip: Mapped[str | None] = mapped_column(Text, nullable=True)
    raw_payload: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    normalized_at: Mapped[datetime] = mapped_column(TIMESTAMP(timezone=True), server_default=func.now(), nullable=False, index=True)
    playbook_triggered: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    playbook_execution_id: Mapped[UUID | None] = mapped_column(PG_UUID(as_uuid=True), nullable=True)
